modules/vmfcore/include/vmf/zlib_compressor.hpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2015 Intel(r) Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http ://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef ZLIB_COMPRESSOR_HPP
+#define ZLIB_COMPRESSOR_HPP
+
+#include "icompressor.hpp"
+
+namespace vmf {
+
+/*!
+ * \class ZLibCompressor
+ * \brief Compression algorithm that uses ZLib library.
+ * Currently runs Deflate algorithm with default settings.
+ */
+class VMF_EXPORT ZLibCompressor : public ICompressor
```

Comment:
I suggest `CompressorXXX` naming (i.e. `CompressorZLib`) to have them all next to each other in a class browser.
Also according to VMF de-facto style the better header file name is `compressorzlib.h`